lib/test-utils/src/Test/Hspec/Extra.hs

```diff
@@ -97,27 +101,32 @@ aroundAll acquire =
         release  <- newEmptyMVar
         done     <- newEmptyMVar
 
-        pid <- async $ do
-            acquire $ \a -> do
-                putMVar resource a
-                await release
-            unlock done
+        pid <- async $ finally (unlock done) $ acquire $ \a -> do
```

Comment:
:+1:

Reply:
Actually, there was a mistake here which broke all tests cleanup. It's so embarrassing, I have force pushed over it.